src/Microsoft.VisualBasic.Core/tests/FileSystemTests.cs

```diff
@@ -723,7 +724,15 @@ public void Write_ArgumentException()
             {
                 ex = e;
             }
-            Assert.NotNull(ex?.GetType() == typeof(TException));
+            if (ex == null)
+            {
+                throw new ThrowsException(typeof(TException));
+            }
+            
+            if (ex.GetType() != typeof(TException))
+            {
+                throw new ThrowsException(typeof(TException), ex);
+            }
```

Comment:
Seems like this could just be `Assert.IsType<TException>(ex);`.

Reply:
Well, that'd change what the failure message looked like.  `ThrowsException` is what Assert.Throws uses internally, it's the exception that is used to build the failure messages the right way.

Reply:
Sure.  Does it matter?  I think all this code should go away anyway (after this PR); it's not clear to me why Assert.Throws can't be used (I see the comment, but don't understand it), and I just tried with eeverything switched to use Assert.Throws and both debug and release passed.  But maybe I'm missing something.

Reply:
> Sure. Does it matter? 

You wound me, sir. :smile:.  The previous one (where exception is null) definitely matters, since it is the "Expected: SomeSortOfException\nActual: No exception".  I think Assert.IsType would result in just saying what type the wrong exception was, but throwing ThrowsException makes it show "it was the wrong type, and here's the full exception".

>  I think all this code should go away anyway

That sounds grand.  But (to me) if it needs to be here it should be correctly modelling the failure messages.

> But maybe I'm missing something.

Maybe it was for netfx runs? (Which Configurations.props says isn't run for this library (anymore))

Reply:
> "it was the wrong type, and here's the full exception"

Fair enough.

> if it needs to be here it should be correctly modelling the failure messages

That's fine.  I just don't think it should be here.

Reply:
> Sure. Does it matter? I think all this code should go away anyway (after this PR); it's not clear to me why Assert.Throws can't be used (I see the comment, but don't understand it), and I just tried with eeverything switched to use Assert.Throws and both debug and release passed. But maybe I'm missing something.
> cc: @cston

Based on this, should I even take any action right now?

Reply:
Nah, just leave what you have. Thanks.